patroni/postgresql.py

```diff
@@ -71,6 +71,32 @@ def null_context():
     yield
 
 
+def _update_postmaster_info(func):
+    def wrapper(self):
+        ret = func(self)
+        if isinstance(ret, dict) and ret:
+            if 'pid' in ret and 'start_time' in ret:
+                old_pid = self._postmaster_info.get('pid', 0)
+                old_start_time = self._postmaster_info.get('start_time', 0)
+                try:
+                    pmpid = int(ret['pid'])
+                    pmstart = int(ret['start_time'])
+                    if pmpid != old_pid or pmstart != old_start_time:
+                        self._postmaster_info = {'pid': pmpid, 'start_time': pmstart}
+                        logger.info("Updated postmaster info {}".format(self._postmaster_info))
+                except ValueError:
+                    logger.exception("Cannot update postmaster info with data due garbage in pid file: {}".format(ret))
```

Comment:
We don't really want to see an entire stack trace here, warning would be more appropriate.
Also logging functions support "%" placeholders, it is not really nice to mix different types of formatting.
Basically `logger.warning('Cannot update postmaster info with data due garbage in pid file: %s', ret)` would be good enough.

Two other log messages are unnecessary:
line 93 -- this one will never happen
line 90 -- it is a normal situation when only part of data is written into postmaster.pid and it represents different stages of postmaster start.

Reply:
If that is the case I suppose that we can remove check `isinstance(ret, dict)` as well because we know return type of `read_pid_file`.

Reply:
Sure, `isinstance(ret, dict)` check is not really needed.

I've just posted here very simplified version.